src/python-bindings/collector.cpp

```diff
@@ -14,6 +14,12 @@
 
 #include "module_lock.h"
 
+// http://docs.python.org/3/c-api/apiabiversion.html#apiabiversion
+#if PY_MAJOR_VERSION >= 3
+   #define PyInt_Check(op)  PyNumber_Check(op)
+   #define PyString_Check(op)  PyBytes_Check(op)
```

Comment:
Yikes.

I don't like having a macro that redefines a python 2 function.  Can you make it really clear this is a macro call?  Maybe something like `PY_COMPAT_INT_CHECK`?

It will definitely cause more source code churn, but seems the right thing to do.